heartbeat/docs/reference/configuration/heartbeat-options.asciidoc

```diff
@@ -0,0 +1,448 @@
+
+
+//Steffen: Can you suggest a realistic path? I have no idea where users would keep these files.  I'm not sure if users specif a glob pattern here or the path to a single file.
+
+*`path`*:: Specifies the path to check for updates. 
+*`interval`*:: Specifies how often Heartbeat checks the file for changes.
+
+[[monitor-icmp-options]]
+==== ICMP Options
+
+These options are valid when the `type` is `icmp`.
+
+[[monitor-icmp-hosts]]
+===== hosts
+
+A list of hosts to ping.
+
+[[monitor-icmp-wait]]
+===== wait
+
+The duration to wait before emitting another ICMP Echo Request. The default is 1
+second (1s).
+
+[[monitor-tcp-options]]
+==== TCP Options
+
+These options are valid when the `type` is `tcp`.
+
+[[monitor-tcp-hosts]]
+===== hosts
+
+A list of hosts to ping. The entries in the list can be:
+
+* A plain host name, such as `localhost`, or an IP address. If you specify this
+option, you must also specify a value for <<monitor-tcp-ports,`ports`>>.  If the
+monitor is configured to use SSL, Heartbeat establishes an SSL/TLS-based
+connection. Otherwise, it establishes a plain TCP connection.
+* A hostname and port, such as `localhost:12345`. Heartbeat connects
+to the port on the specified host. If the monitor is configured to use SSL,
+Heartbeat establishes an SSL/TLS-based connection. Otherwise, it establishes
+a TCP connection.
+* A full URL using the syntax `scheme://<host>:[port]`, where:
+** `scheme` is one of `tcp`, `plain`, `ssl` or `tls`. If `tcp` or `plain` is
+specified, Heartbeat establishes a TCP connection even if the monitor is
+configured to use SSL. If `tls` or `ssl` is specified, Heartbeat establishes
+an SSL connection. However, if the monitor is not configured to use SSL, the
+system defaults are used (not supported on Windows).
+** `host` is the hostname.
+** `port` is the port number. If `port` is missing in the URL, the
+<<monitor-tcp-ports,`ports`>> setting is required.
+
+// Questions: Can port above be a list of ports, or is it only a single port.
+
+[[monitor-tcp-ports]]
+===== ports
+
+A list of ports to ping if the host specified in <<monitor-tcp-hosts,`hosts`>>
+does not contain a port number.
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: tcp 
+  schedule: '@every 5s'
+  hosts: ["myhost"]
+  ports: [80, 9200, 5044]
+  ipv4: true
+  ipv6: true
+  mode: any
+  timeout: 16s
+-------------------------------------------------------------------------------
+
+[[monitor-tcp-check]]
+===== check
+
+An optional payload string to send to the remote host and the expected answer.
+If no payload is specified, the endpoint is assumed to be available if the
+connection attempt was successful. If `send` is specified without `receive`,
+any response is accepted as OK. If `receive` is specified without `send`, no
+payload is sent, but the client expects to receive a payload on connect.
```

Comment:
@urso Or did you mean to say here that the optional payload that's sent is a "hello message" or "banner"?